packages/react-native-usb/src/index.ts

```diff
@@ -0,0 +1,144 @@
+    const perf = performance.now();
+    const data = await ReactNativeUsbModule.transferIn(deviceName, endpointNumber, length)
+        .catch((error: any) => {
+            debugLog('JS: USB read error: ', error);
+            throw error;
+        })
+        // eslint-disable-next-line arrow-body-style
+        .then((result: number[]) => {
+            // debugLog('JS: Native USB read result:', JSON.stringify(result));
+
+            return {
+                data: new Uint8Array(result),
+                status: 'ok',
+            };
+        });
+    debugLog('JS: USB read time', performance.now() - perf);
+
+    return data;
+};
+
+const transferOut = async (
+    deviceName: string,
+    endpointNumber: number,
+    data: Uint8Array | BufferSource,
+) => {
+    try {
+        const perf = performance.now();
+        await ReactNativeUsbModule.transferOut(deviceName, endpointNumber, data.toString());
+        debugLog('JS: USB write time', performance.now() - perf);
+        return { status: 'ok' };
+    } catch (error) {
+        debugLog('JS: USB write error', error);
+        throw error;
+    }
+};
+
+// eslint-disable-next-line require-await
+const createNoop = (methodName: string) => async () => {
+    debugLog(`Calling ${methodName} which is not implemented.`);
+};
+
+const createWebUSBDevice = (device: NativeDevice): WebUSBDevice => ({
+    ...device,
+    open: () => open(device.deviceName),
+    close: () => close(device.deviceName),
+    forget: createNoop('forget'),
+    selectConfiguration: (configurationValue: number) =>
+        selectConfiguration(device.deviceName, configurationValue),
+    claimInterface: (interfaceNumber: number) => claimInterface(device.deviceName, interfaceNumber),
+    releaseInterface: (interfaceNumber: number) =>
+        releaseInterface(device.deviceName, interfaceNumber),
+    selectAlternateInterface: createNoop('selectAlternateInterface'),
+    controlTransferIn: createNoop('controlTransferIn'),
+    controlTransferOut: createNoop('controlTransferOut'),
+    clearHalt: createNoop('clearHalt'),
+    transferIn: (endpointNumber: number, length: number) =>
+        transferIn(device.deviceName, endpointNumber, length),
+    transferOut: (endpointNumber: number, data: BufferSource) =>
+        transferOut(device.deviceName, endpointNumber, data),
+    isochronousTransferIn: createNoop('isochronousTransferIn'),
+    isochronousTransferOut: createNoop('isochronousTransferOut'),
+    reset: createNoop('reset'),
+
+    // TODO: Implement these properties
+    usbVersionMajor: 2,
+    usbVersionMinor: 0,
+    usbVersionSubminor: 0,
+    deviceVersionMajor: 1,
+    deviceVersionMinor: 0,
+    deviceVersionSubminor: 0,
+    configurations: [],
+});
+
+export function onDeviceConnected(listener: (event: OnConnectEvent) => void): Subscription {
+    return emitter.addListener<NativeDevice>('onDeviceConnect', event => {
```

Comment:
Are these listeners cleared anywhere?

Reply:
No, this is purely proxied to "client"

Reply:
There should be probably something like removeListeners but since tranport/connect doesnt use them anyway I did not implement it